programs/relayer/src/modules/cosmos_to_eth.rs

```diff
@@ -68,21 +92,38 @@ impl CosmosToEthRelayerModuleService {
             .unwrap_or_else(|e| panic!("failed to create provider: {e}"));
 
         let eth_listener = eth_eureka::ChainListener::new(config.ics26_address, provider.clone());
-        let submitter = if config.mock {
-            TxBuilder::new_mock(config.ics26_address, provider, tm_client)
-        } else {
-            TxBuilder::new(
-                config.ics26_address,
-                provider,
-                tm_client,
-                config.sp1_private_key,
-            )
+
+        let sp1_prover: Box<dyn Prover<CpuProverComponents>> = match config.sp1_config {
```

Comment:
Why is it CpuProverComponents? Looks weird with the choices below, but this is how it works?

Reply:
Am I wrong to think it ought to be `SP1ProverComponents`

Reply:
SP1ProverComponents is an interface, whereas the match arm are actual implementations. They chose to implement `Prover<CpuProverComponents>` for all of them. Complain to sp1 lol

Reply:
lol, that is what I was afraid of 😂 fair enough!